lib/rules/no-unneeded-ternary.js

```diff
@@ -134,7 +134,20 @@ module.exports = {
                         node,
                         loc: node.consequent.loc.start,
                         message: "Unnecessary use of conditional expression for default assignment.",
-                        fix: fixer => fixer.replaceText(node, `${astUtils.getParenthesisedText(sourceCode, node.test)} || ${astUtils.getParenthesisedText(sourceCode, node.alternate)}`)
+                        fix: fixer => {
+                            let nodeTest = astUtils.getParenthesisedText(sourceCode, node.test);
+                            let nodeAlternate = astUtils.getParenthesisedText(sourceCode, node.alternate);
+
+                            if (node.test.type === "ConditionalExpression") {
+                                nodeTest = `(${nodeTest})`;
+                            }
```

Comment:
This is similar to @platinumazure's comment, but I'm not sure checking for `node.test` is necessary here. If it is necessary, please add a test for it; otherwise, it's fine to remove it.